### PR TITLE
Add warning about eigenvalues of Ptilde

### PR DESCRIPTION
**Pull request by cc7768**
Kills asset pricing if the primitives do not satisfy that the modulus of the eigenvalues of Ptilde are less than 1.

### CODE REVIEW

File: markov_asset/asset_pricing.py [hunk 1]
```diff
@@ -82,6 +82,14 @@ def __init__(self, beta, P, s, gamma):
         self.P, self.s = P, s
         self.n = self.P.shape[0]
 
+        # Create P_tilde
+        Ptilde = self.P_tilde
+        ev = eigvals(beta*Ptilde)
+        lt1 = np.all(np.absolute(ev) < 1.0)
```
Comment (oyamad) [inline markov_asset/asset_pricing.py]: Minor comment:
`Ptilde` is a nonnegative matrix, so its dominant eigenvalue is real, and hence it is sufficient to compare the real part:

```python
lt1 = np.all(ev.real < 1.0)
```

In fact, it seems even without `real` the real part is compared:

```python
lt1 = np.all(ev < 1.0)
```

File: markov_asset/asset_pricing.py [hunk 2]
```diff
@@ -82,6 +82,14 @@ def __init__(self, beta, P, s, gamma):
         self.P, self.s = P, s
         self.n = self.P.shape[0]
 
+        # Create P_tilde
+        Ptilde = self.P_tilde
+        ev = eigvals(beta*Ptilde)
```
Comment (oyamad) [inline markov_asset/asset_pricing.py]: Very minor comment:
Here multiplication occurs nxn times, whereas if you do `ev = eigvals(Ptilde) * beta` it's n times.

File: markov_asset/asset_pricing.py [hunk 3]
```diff
@@ -82,6 +82,14 @@ def __init__(self, beta, P, s, gamma):
...
+        lt1 = np.all(np.absolute(ev) < 1.0)
+        if not lt1:
+            msg = "All eigenvalues of Ptilde must be less than 1"
```
Comment (oyamad) [inline markov_asset/asset_pricing.py]: "less than 1/beta"?

**Comment by mmcky** [comment]
@cc7768 Does this PR need to be updated based on the feedback above?

**Comment by cc7768** [comment]
Yes. I had forgotten about this -- Will fix and update today.

Thanks @oyamad and @mmcky

**Comment by spencerlyon2** [comment]
Ping! Let's get to 0 PRs!

**Comment by cc7768** [comment]
I addressed each of @oyamad comments -- Thank you for the feedback.

Also rebased wth master and made sure that the notebook ran as well (using my version not the master branch version -- I left the notebook as is though, so that when it is merged everything should just work).

**Pull request merged**